core/src/main/java/com/google/cloud/sql/core/InstanceDataSupplier.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.sql.core;
+
+import com.google.cloud.sql.AuthType;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.google.common.util.concurrent.ListeningScheduledExecutorService;
+import java.security.KeyPair;
+import java.util.concurrent.ExecutionException;
+
+/** Internal Use Only: Gets the instance data for the CloudSqlInstance from the API. */
+interface InstanceDataSupplier {
+  /**
+   * Internal Use Only: Gets the instance data for the CloudSqlInstance from the API.
+   *
+   * @param instanceName the name of the instance
```

Comment:
If we don't have a linter complaining about these, I'd suggest leaving off params. The types are clear enough and the docs don't provide much value in my opinion.

Reply:
Fixed.